native/core/src/common/bit.rs

```diff
@@ -145,7 +145,9 @@ macro_rules! read_num_be_bytes {
 #[inline]
 pub fn memcpy(source: &[u8], target: &mut [u8]) {
     debug_assert!(target.len() >= source.len(), "Copying from source to target is not possible. Source has {} bytes but target has {} bytes", source.len(), target.len());
-    target[..source.len()].copy_from_slice(source)
+    // Originally `target[..source.len()].copy_from_slice(source)`
+    // We use the unsafe copy method to avoid some expensive bounds checking/
+    unsafe { std::ptr::copy_nonoverlapping(source.as_ptr(), target.as_mut_ptr(), source.len()) }
```

Comment:
`copy_from_slice` is `copy_nonoverlapping` with the `if self.len() != src.len()` check.
Do you have any benchmark results to show the difference by any chance?

Reply:
Before this 
`scan decimal (spark) : 1.0x`
`scan decimal (Comet, decimal 128) : 0.4x`
After
`scan decimal (Comet, decimal 128) : 0.5x`
So a small improvement.